### ISSUE TITLE
Warn user if strategy is DDP but seed_everything() has not been called

### ISSUE THREAD

**Issue by samgelman** [PyTorchLightning/pytorch-lightning#12986]
## 🚀 Feature

Warn user if strategy is DDP but seed_everything() has not been called

### Motivation

Initial weights might be different on each device if seed_everything() is not called. Need to call seed_everything() to get correct DDP results.

### Pitch

A simple solution would be to throw a warning if the strategy is DDP but seed_everything() has not been called. Can also update the documentation to state seed_everything() should be called for DDP. 

### Alternatives

- Automatically seed_everything() when strategy is DDP (letting user know a seed has been set due to DDP)
- Update documentation to state that seed_everything() should be called for DDP. Could do it [here](https://pytorch-lightning.readthedocs.io/en/latest/accelerators/gpu_intermediate.html) or [here](https://pytorch-lightning.readthedocs.io/en/latest/accelerators/accelerator_prepare.html)

### Additional context

I forgot to set a seed when using DDP. Would be great to get a reminder either as a warning or in the docs.
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

**Comment by ananthsub** [comment]
This shouldn't be needed. DistributedDataParallel internally broadcasts module states from rank 0 to synchronize weights.

**Comment by ananthsub** [comment]
> If that's true

Here is the code pointer for how thi sis handled within DistributedDataParallel in particular: https://github.com/pytorch/pytorch/blob/56bed0dcfe7ca9047e5c95a6f3d7fcb0ec403b0c/torch/nn/parallel/distributed.py#L648-L655

If using another library, seeding may still be needed (depending on if that library syncs the initial states).

**Comment by samgelman** [comment]
Got it, thank you!
